dace/frontend/python/replacements.py

```diff
@@ -4305,19 +4353,21 @@ def reshape(pv: 'ProgramVisitor',
     else:
         strides = [data._prod(newshape[i + 1:]) for i in range(len(newshape))]
 
-    newarr, _ = sdfg.add_view(arr,
-                              newshape,
-                              desc.dtype,
-                              storage=desc.storage,
-                              strides=strides,
-                              allow_conflicts=desc.allow_conflicts,
-                              total_size=desc.total_size,
-                              may_alias=desc.may_alias,
-                              alignment=desc.alignment,
-                              find_new_name=True)
+    newarr, newdesc = sdfg.add_view(arr,
+                                    newshape,
+                                    desc.dtype,
+                                    storage=desc.storage,
+                                    strides=strides,
+                                    allow_conflicts=desc.allow_conflicts,
+                                    total_size=desc.total_size,
+                                    may_alias=desc.may_alias,
+                                    alignment=desc.alignment,
+                                    find_new_name=True)
 
     # Register view with DaCe program visitor
-    pv.views[newarr] = arr
+    aset = subsets.Range.from_array(desc)
+    vset = subsets.Range.from_array(newdesc)
+    pv.views[newarr] = (arr, Memlet(f'{arr}[{aset}]->{vset}'))
```

Comment:
would be cleaner and safer to use `Memlet(data=arr, subset=aset, other_suset=vset)`

Reply:
I thought that we actually preferred the string representation but it is easy to swap.

Reply:
No, we prefer the raw objects as much as possible (faster performance, less ambiguity). The strings are mostly for convenience.